packages/micropip/test_micropip.py

```diff
@@ -236,3 +236,22 @@ def test_install_mixed_case2(selenium_standalone_micropip, jinja2):
         `);
         """
     )
+
+
+def test_report_all_failed_dependencies():
+    pytest.importorskip("packaging")
+    from micropip import micropip
+
```

Comment:
Could you try [monkeypatching](https://docs.pytest.org/en/6.2.x/monkeypatch.html) the http request to return the expected JSON response for this test? Unless it ends up being too complicated. You don't need to return the full JSON, just the fields that are used the dependency resolution. It doesn't need to be pytest/tensorflow could be an arbitrary fake package.

I know we have a few tests that make HTTP calls to PyPi, but it's really not ideal to do this systematically.

Reply:
It could look approximately like the following,
```py
async def _mock_fetch_string(url, **kwargs):
    if url.endswith("package1"):
        return "JSON encoded as a string"
    elif url.endswith("package1")
       return "another JSON"

def test_report_all_failed_dependencies(monkeypatch):
    pytest.importorskip("packaging")
    monkeypatch.setattr('micropip.micropip.fetch_string', _mock_fetch_string)
    ...
```

The returned  JSON can be similar aside from the few relevant fields that matter for this test